### ISSUE TITLE
Add rule for files that has Byte Order Mark (BOM) character

### ISSUE THREAD

**Issue by sampeson** [MarketSquare/robotframework-robocop#316]
Notepad and some other editors can create text files with [BOM](https://en.wikipedia.org/wiki/Byte_order_mark) that can affect Robot test execution. E.g. with vscode choose `UTF-8 with BOM` encoding and save file.

When run robocop against it it complains like this (this is run on osx):
```
$ robocop bom_example.robot 
/Users/sapesonen/Desktop/bom_example.robot:1:0 [W] 0704 Ignored data found in file
```

On Windows 10 it did not complain like this. On Windows server it complains so your mileage may vary.

See example file:
[bom_example.robot.zip](https://github.com/MarketSquare/robotframework-robocop/files/6244973/bom_example.robot.zip)

So propose to create separate rule to find these kind of problems so that can fix the encoding to UTF-8.

**Comment by bhirsz** [comment]
I see that this task splits into two: 
1) don't complain about 0704 if it's only BOM
2) Detect BOM and report it (on every os) but using different rule

I will check how it behaves using your files first
